### ISSUE TITLE
Improve ErrorHandler performance 

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#8253]
From this JIRA ticket:
https://its.cern.ch/jira/browse/CMSCOMPPR-1556

we see it can handle around 5k jobs/h, jobs done with all retries. It's not going to be simple, since there are lots of actions involved, but we should eventually look at this code and make it more efficient.

**Comment by amaltaro** [comment]
While working on another issue on the same component, I think we can get a reasonable speed up if we sort `failedJobs` by workflow and task. It will avoid lots of CouchCollection and CouchFileset creations, see
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/ACDC/DataCollectionService.py#L90

to profit of such changes, we need then increase the number of failed jobs processed per cycle (or slice).

**Comment by ticoann** [comment]
@amaltaro, Do you think is this anything related memory issue. I see performance increased when we use common   object for lumi, since we don't have to load again. But it could be the totally different case.

**Comment by amaltaro** [comment]
Yes, memory usage was probably affecting the performance of this component too. However, I think the worst offenders are related to (from the top of my mind, I might be slightly wrong :):
*) how we handle each job, building objects back and forth and loading a bunch of information for each of them
*) how we create the ACDC-wise objects and upload them to acdcserver (this one should be better now, but we still have to let the component handle >= 500 jobs in one shot, otherwise there is no gain.

Now that we fixed the memory consumption, I suggest we update the ErrorHandler maxProcessSize to 500 and start observing the component performance (having this info in grafana would be useful... I'll try to get to that..).
